### PR TITLE
pegtl 3.0.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/pegtl.rb [hunk 1]
```diff
@@ -14,20 +14,23 @@ class Pegtl < Formula
   end
 
   depends_on "cmake" => :build
+  # Fix `error: 'path' is unavailable: introduced in macOS 10.15`
+  depends_on macos: :catalina
```
Comment (SMillerDev) [inline Formula/pegtl.rb:18]: Shouldn't that be catalina or higher?

Comment (chenrui333) [reply Formula/pegtl.rb]: yeah, you are right, I was confused about the right syntax.

Let me try some other solution for solving Mojave build. :)

File: Formula/pegtl.rb [hunk 2]
```diff
@@ -15,19 +15,27 @@ class Pegtl < Formula
 
   depends_on "cmake" => :build
 
+  if MacOS.version <= :mojave
+    depends_on "gcc"
+    fails_with :clang do
+      cause "'path' is unavailable: introduced in macOS 10.15"
```
Comment (fxcoudert) [inline Formula/pegtl.rb:21]: The error message about path is not very clear to me, maybe add `C++17` somewhere so people understand better?

Comment (chenrui333) [reply Formula/pegtl.rb:21]: I actually directly adopt the same change from somewhere. Can you make a code suggestion, I can adopt it.

Comment (chenrui333) [reply Formula/pegtl.rb:21]: ping @fxcoudert :)

Comment (fxcoudert) [reply Formula/pegtl.rb:21]: Sorry, spending all my time trying to make CI nodes run again. It's a minor thing, do as you wish.

Comment (SMillerDev) [inline Formula/pegtl.rb:21]:
```suggestion
      cause "'path' is unavailable in c++ < 17: introduced in macOS 10.15"
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**